### PR TITLE
feat: PerunEnsureMember

### PR DESCRIPTION
**Pull request by Br3nKo**

### CODE REVIEW

File: microservices/perun_ensure_member.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+        if self.__callback_param_name:
+            registration_url = self.__register_url + '?vo=' \
+                               + self.__vo_short_name
+            if group_name:
+                registration_url += '&group=' + group_name
+
+            params[PerunConstants.TARGET_NEW] = callback
+            params[PerunConstants.TARGET_EXISTING] = callback
+            params[PerunConstants.TARGET_EXTENDED] = callback
```
Comment (melanger) [inline microservices/perun_ensure_member.py:181]: similar code is in https://github.com/CESNET/satosacontrib.perun/pull/8, please consult with @PeterBolha and extract it into utils

**Review by melanger — CHANGES_REQUESTED** [review]
please (later) add config example and unit tests

**Comment by Br3nKo** [comment]
Checks will not be successfull until Utils and PerunConstants are merged

**Review by melanger — APPROVED** [review]

File: satosacontrib/perun/micro_services/perun_ensure_member.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                'before this microservice?'
+            )
+        vo = self.__adapters_manager.get_vo(short_name=self.__vo_short_name)
+        if not vo:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'VO with vo_short_name \''
+                + self.__vo_short_name + '\' not found.'
+            )
+
+        self.__handle_user(user, vo, data, context)
```
Comment (xpavlic) [inline satosacontrib/perun/micro_services/perun_ensure_member.py:99]: We do not save user in data anymore. We just save user_id into attributes. Change user variables to user_id.
```suggestion
        user_id = data.attributes.get(self.__global_cfg["perun_user_id_attribute"])
        if not user_id:
            raise SATOSAError(
                self.LOG_PREFIX + 'Missing mandatory field '
                                  '\'perun.user\' in request. Hint: Did you '
                                  'configured PerunIdentity microservice '
                                  'before this microservice?'
            )
        vo = self.__adapters_manager.get_vo(short_name=self.__vo_short_name)
        if not vo:
            raise SATOSAError(
                self.LOG_PREFIX + 'VO with vo_short_name \''
                + self.__vo_short_name + '\' not found.'
            )

        self.__handle_user(user_id, vo, data, context)
```

File: requirements.txt [hunk 3]
```diff
@@ -1 +1,10 @@
-.
\ No newline at end of file
+SATOSA~=8.1.0
```
Comment (xpavlic) [inline requirements.txt:1]: revert changes

**Review by xpavlic — CHANGES_REQUESTED** [review]

**Comment by xpavlic** [comment]
Rebase branch and pipelines will pass, update tests.

File: satosacontrib/perun/config_templates/perun_ensure_member.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module: satosacontrib.perun.micro_services.perun_ensure_member.PerunEnsureMember
```
Comment (xpavlic) [inline satosacontrib/perun/config_templates/perun_ensure_member.yaml:1]: change paremters to snake_case (also in code)

**Review by xpavlic — CHANGES_REQUESTED** [review]

**Review by xpavlic — APPROVED** [review]

**Pull request merged**